src/main/java/com/internship/socialnetwork/service/impl/PostServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package com.internship.socialnetwork.service.impl;
+
+import com.internship.socialnetwork.dto.PostDto;
+import com.internship.socialnetwork.exception.NotFoundException;
+import com.internship.socialnetwork.model.Post;
+import com.internship.socialnetwork.repository.PostRepository;
+import com.internship.socialnetwork.service.PostService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class PostServiceImpl implements PostService {
+
+    private final PostRepository postRepository;
+
+    @Override
+    public Post getModel(Long id) {
+        return postRepository.findById(id).orElseThrow(() -> new NotFoundException("Post with id: " + id + " not found!"));
+    }
+
+    @Override
+    public List<PostDto> getAll() {
+        return postRepository.findAll().stream().map(PostDto::convertToDto).toList();
+    }
+
+    @Override
+    public List<PostDto> getAllByAuthorId(Long id) {
+        return postRepository.findByAuthor(id).stream().map(PostDto::convertToDto).toList();
+    }
+
+    @Override
+    public PostDto get(Long id) {
+        return PostDto.convertToDto(postRepository.findById(id).orElseThrow(() -> new NotFoundException("Post with id: " + id + " not found!")));
+    }
+
+    @Override
+    public PostDto create(PostDto postDto) {
+        Post post = postDto.convertToModel();
+        return PostDto.convertToDto(postRepository.save(post));
+    }
+
+    @Override
+    public PostDto update(Long id, PostDto postDto) {
+        Post post = getModel(id);
+        post.setText(postDto.getText());
+        post.setVideoPath(postDto.getVideoPath());
+        post.setImagePath(postDto.getImagePath());
+        return PostDto.convertToDto(postRepository.save(post));
+    }
+
+    @Override
+    public void delete(Long id) {
+        postRepository.findById(id).orElseThrow(() -> new NotFoundException("Post with id: " + id + " not found!"));
```

Comment:
you don't need to search by id 2x, the method below also does find by id. You can put postRepository.delete(getModel(id)) instead